Adding New rmsynthesis worker

Mulan-94 opened the pull request:
- This an attempt to interface `rmsynthesis` to the `polimager` worker

SpheMakh commented:
ok to test

francescaLoi commented:
Let's wait until the #1479 is merged otherwise the workflow will be interrupted.

Mulan-94 committed:
Adds new RM sythesis worker
Updating from 2023

Mulan-94 committed:
Fix some output routing

Mulan-94 committed:
Stack single images before rmsynthesis

Mulan-94 committed:
Ruff formatting checks included

Mulan-94 committed:
More ruff

Mulan-94 committed:
Add more user options for rmsynthesis and rmclean
Ignore provided stokes V because rmtools does not accept them as input
Add a use_i option to use stokes I

Mulan-94 committed:
Ruff stuff

francescaLoi commented:
can you please provide a simple config file for the new worker?

Mulan-94 commented:
> can you please provide a simple config file for the new worker?

@francescaLoi yes, you can try this one?

```
rmsynthesis:
  enable: true
  label_in: ""
  cubes:
    enable: false
    # i_cube: output-2023-old/output/continuum/polarization/i-conv-image-cube.fits
    # q_cube: output-2023-old/output/continuum/polarization/q-conv-image-cube.fits
    # u_cube: output-2023-old/output/continuum/polarization/ul-conv-image-cube.fits
  singles:
    enable: true
    i_prefix: output/continuum/polarization/ugali-2026_PictorA-[0-9][0-9][0-9][0-9]-I-*convolved.fits 
    q_prefix: output/continuum/polarization/ugali-2026_PictorA-[0-9][0-9][0-9][0-9]-Q-*convolved.fits
    u_prefix: output/continuum/polarization/ugali-2026_PictorA-[0-9][0-9][0-9][0-9]-U-*convolved.fits
    v_prefix: output/continuum/polarization/ugali-2026_PictorA-[0-9][0-9][0-9][0-9]-V-*convolved.fits
  # freq_file: output/continuum/polarization/frequencies.txt
  max_phi: 200
  n_samples: 10
  use_i: false
  ncpus: 3
  cutoff: 1
  freq_file: freqs.dat
  max_iter: 100
  prefix: tentacles


```

francescaLoi commented:
What is happening?
```
2026-04-15 13:41:55 CARACal INFO: Invoked as /home/floi/.local/bin/caracal -ct singularity -c rmsynth.yml
Traceback (most recent call last):
  File "/home/floi/.local/lib/python3.10/site-packages/caracal/main.py", line 234, in main
    parser.populate_parser(config)
  File "/home/floi/.local/lib/python3.10/site-packages/caracal/dispatch_crew/config_parser.py", line 191, in populate_parser
    self._process_subparser_tree(variables, self._schemas[worker][1], base_section=worker)
  File "/home/floi/.local/lib/python3.10/site-packages/caracal/dispatch_crew/config_parser.py", line 312, in _process_subparser_tree
    default_value = subVars["example"]
KeyError: 'example'
2026-04-15 13:41:55 CARACal ERROR: Error parsing arguments or configuration: 'example'
```
These are the worker parameters:
```
rmsynthesis:
  enable: true
  label_in: "spw"
  cubes:
    enable: true
  singles:
    enable: false
  max_phi: -1000
  n_samples: 10
  use_i: false

```
